src/arrayMethodJoin.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let sep = String(separator);
+
+    if (separator === undefined) {
+      sep = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
+      }
+
+      if (i === this.length - 1) {
+        sep = '';
+      }
+      result += `${this[i] + sep}`;
```

Comment:
```suggestion
      if (this[i] !== undefined || this[i] !== null) {
        result += this[i];
      }

      if (i !== this.length - 1) {
        result += sep
      }
```

Reply:
Tests don't pass with this code. I changed it another way with mentors' suggestions.